docs-ui/stories/components/compositeSelect.stories.js

```diff
@@ -0,0 +1,66 @@
+import CompositeSelect from 'sentry/components/compactSelect/composite';
+
+export default {
+  title: 'Components/Composite Select',
+  component: CompositeSelect,
+};
+
+export const _CompositeSelect = props => (
+  <CompositeSelect {...props}>
+    <CompositeSelect.Region
+      label="Group 1"
+      defaultValue="choice_one"
+      onChange={() => {}}
+      options={[
+        {value: 'choice_one', label: 'Choice One'},
+        {value: 'choice_two', label: 'Choice Two'},
+      ]}
+    />
+    <CompositeSelect.Region
+      multiple
+      label="Group 2"
+      defaultValue={['choice_three']}
+      onChange={() => {}}
+      options={[
+        {value: 'choice_three', label: 'Choice Three'},
+        {value: 'choice_four', label: 'Choice Four'},
+      ]}
+    />
+  </CompositeSelect>
```

Comment:
`CompositeSelect` now has a composable structure, making it more clear that each "region" has a separate selection state, and allowing for better type inference.